pxr/base/tf/testenv/stringUtils.cpp

```diff
@@ -386,6 +395,35 @@ TestStrings()
     TF_AXIOM(TfUnstringify<char>("a") == 'a');
     TF_AXIOM(TfStringify("string") == "string");
     TF_AXIOM(TfUnstringify<string>("string") == "string");
+    TF_AXIOM(TfStringify(1000) == "1000");
+    
+    // make sure we can represent the min and max of each type
+    TF_AXIOM(_RoundtripStringifyLimits<short>());
+    TF_AXIOM(_RoundtripStringifyLimits<int>());
+    TF_AXIOM(_RoundtripStringifyLimits<long>());
+    TF_AXIOM(_RoundtripStringifyLimits<long long>());
+    TF_AXIOM(_RoundtripStringifyLimits<unsigned short>());
+    TF_AXIOM(_RoundtripStringifyLimits<unsigned int>());
+    TF_AXIOM(_RoundtripStringifyLimits<unsigned long>());
+    TF_AXIOM(_RoundtripStringifyLimits<unsigned long long>());
+    
+    // verify that TfStringify is agnostic to locale for
+    // numerical values
+    std::locale originalLocale;
+    std::locale::global(std::locale(""));
+    try
+    {
+        TF_AXIOM(TfStringify(1000.56) == "1000.56");
+        TF_AXIOM(TfStringify(1000) == "1000");
+        std::locale::global(originalLocale);
```

Comment:
This new test did not fail for me when I ran it without the `TfStringify` fix. I'm guessing that your environment might have a locale set that would cause this to fail?

We could install a facet that would force formatting for integral types into the locale to make this independent of environment. With the below setup, the test fails without the `TfStringify` fix and passes with it.

```
struct separate_thousands : std::numpunct<char>
{
    std::string do_grouping() const override { return "\003"; }
    char do_thousands_sep() const override { return ','; }
};
                                                                                                                                                                                                                                     
std::locale originalLocale;
std::locale::global(std::locale(std::locale(""), new separate_thousands));
...
```

Reply:
Probably a difference between Windows and Linux.  I suspect Linux the default native locale on Linux is probably "C" (which would be the same as the locale the program is started with)